File: algorithms/ar-qsrt/golang/bubble_sort.go
package main

import "fmt"

// BubbleSort sorts an array by using an optimized version of the Bubble Sort algorithm
func BubbleSort(arr []int) {
	isSorted := false
	n := len(arr)

	for !isSorted {
		isSorted = true
		for i := 0; i < n-1; i++ {
			if arr[i+1] < arr[i] {
				temp := arr[i+1]
				arr[i+1] = arr[i]
				arr[i] = temp
				isSorted = false
			}
		}
	}

}

func main() {
	testArr := []int{2, 4, 1, 12, 7, 0, 3, 14, 19, 11}
	fmt.Printf("%v\n", testArr)
	BubbleSort(testArr)
	fmt.Printf("%v\n", testArr)
}

File: algorithms/ar-binsrt/c/Bubble Sort.c
#include<stdio.h>
//function for swapping 2 numbers
void swap(int *xp,int *yp)
{
    int temp;
    temp=*xp;
    *xp=*yp;
    *yp=temp;

}
//function for printing the array
void printArray(int arr[],int size)
{
    int i;
    printf("***Sorted array is**\n");
    for(i=0;i<size;i++)
    {

        printf("%d\n",arr[i]);
    }

    printf("\n");
}


//main logic for bubble sort
void bubble_sort(int arr[],int n)
{
    int i,j;
    for(i=0;i<n-1;i++)
    {
        for(j=0;j<n-i-1;j++)
        {
            if(arr[j]>arr[j+1])
                swap(&arr[j],&arr[j+1]);
        }
    }

}


//the main driver code
int main()
{
    int i,arr[50],n;
    printf("Enter array size\n");
    scanf("%d",&n);
    printf("Enter array\n");

    for(i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }

    bubble_sort(arr,n);//calling bubble_sort

    printArray(arr,n);//calling printArray
    printf("\nArray is sorted");
    return 0;

}
